fix handling of empty strings in boolean transforms/condition

jason2e opened the pull request:
BooleanIsNullTransform just checks == null, not "" (BooleanIsNullCondition is fine though)

BooleanRegexTransform and BooleanRegexCondition both automatically go false for blank strings, so behavior is incorrect when the regex is for instance `^$` or `^ *$`.

Commit:
fix handling of empty strings in boolean transforms/condition

katherinebernstein commented on src/main/java/conditions/BooleanRegexCondition.java:
```diff
@@ -51,9 +51,6 @@ public Boolean getReverse() {
     @Override
     public Boolean apply( Map<String, String> row ) {
         String o = row.get( column );
-        if ( StringUtils.isBlank( o ) ) {
```
`Cached.getInsensitiveMatcherForString` includes a `@NotNull` annotation for the `targetString` param. This section catches the case where `o` is null, but without this check it becomes possible to pass a null value to `getInsensitiveMatcherForString`, which is illegal. So you probably either want to keep this or replace it will an explicit null check if you want to allow empty strings.

jason2e replied:
Ooh, ok. Will fix.

jason2e replied:
Now explicitly converts `null` to `""`.

katherinebernstein commented on src/main/java/transforms/BooleanIsNullTransform.java:
```diff
@@ -45,8 +45,8 @@ public boolean applyCondition( Map<String, Object> row ) {
-      
-        return row.get( column ) == null ;
+
+        return row.get( column ) == null || row.get( column ).equals( "" ) ;
```
It's a little counterintuitive that nulls and also empty strings get the same treatment given the class name -- it would probably be clearer to rename something like `IsEmptyTransform` or something like that. 

Also -- I assume you want to treat `""` as empty but not `" "`? If you do, you should probably just use `StringUtils.isBlank( row.get( column ))`;

jason2e replied:
True, IsEmptyTransform would be clearer. BooleanIsNullCondition was already checking for blanks (which is what we need it for), so I just decided to be consistent with that.

jason2e replied:
update: now using isBlank for this. As far as renaming, I'd like to check in with Joke first - I believe she has her own plans in the works for restructuring transforms/conditions.

Commit:
add requested changes and update transform test

Transform test now applies BooleanRegexCondition to a null column using regex "^$", expecting true.

geekbeast committed:
Merge branch 'develop' into INTEGRATE-379
